File: tms-admin-web-ui/src/app/reports/models/dispatch-day-report-row.ts
export interface DispatchDayReportRow {
  dispatchId: number;
  planDate: string; // ISO date (yyyy-MM-dd)
  truckNo: string | null;
  truckTrip: any | null;
  depot: string | null;
  numberOfPallets: number | null;
  truckType: string | null;
  factoryDeparture: string | null; // ISO instant
  depotArrival: string | null; // ISO instant
  plannedDepotArrival: string | null; // ISO instant
  unloadingComplete: string | null; // ISO instant
  finalDestinationText: string | null;
}

File: tms-core-api/src/main/java/com/svtrucking/logistics/modules/reports/dto/DispatchDayReportRow.java
package com.svtrucking.logistics.modules.reports.dto;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;

public record DispatchDayReportRow(
    Long dispatchId,
    LocalDate planDate,
    String truckNo,
    String truckTrip,
    String depot,
    BigDecimal numberOfPallets,
    String truckType,
    Instant factoryDeparture,
    Instant depotArrival,
    Instant plannedDepotArrival,
    Instant unloadingComplete,
    String finalDestinationText) {}
